3741/1/top_gems.rb

```diff
@@ -0,0 +1,31 @@
+require 'yaml'
+require 'optparse'
+require_relative 'lib/yaml_reader'
+require_relative 'lib/gem_data_reader'
+require_relative 'lib/gems_view'
+
+# main run class
+class TopGems
+  FILE_NAME = 'gems.yml'.freeze
+  # :reek:FeatureEnvy
```

Comment:
# :reek:FeatureEnvy
means that you need a separate class

you could create  MyOptionParser with interface like this
```
def options
  @options ||= MyOptionParser.new.get_options
end
```
and then use:
options[:top]
options[:name]